src/panels/config/script/ha-script-field-row.ts

```diff
@@ -0,0 +1,221 @@
+import { ActionDetail } from "@material/mwc-list/mwc-list-foundation";
+import "@material/mwc-list/mwc-list-item";
+import { mdiDelete, mdiDotsVertical } from "@mdi/js";
+import { CSSResultGroup, LitElement, css, html } from "lit";
+import { customElement, property } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { fireEvent } from "../../../common/dom/fire_event";
+import "../../../components/ha-alert";
+import "../../../components/ha-button-menu";
+import "../../../components/ha-card";
+import "../../../components/ha-expansion-panel";
+import "../../../components/ha-icon-button";
+import { Field } from "../../../data/script";
+import { showConfirmationDialog } from "../../../dialogs/generic/show-dialog-box";
+import { haStyle } from "../../../resources/styles";
+import type { HomeAssistant } from "../../../types";
+import type { SchemaUnion } from "../../../components/ha-form/types";
+
+const SCHEMA = [
+  {
+    name: "name",
+    selector: { text: {} },
+  },
+  {
+    name: "description",
+    selector: { text: {} },
+  },
+  {
+    name: "required",
+    selector: { boolean: {} },
+  },
+  {
+    name: "example",
+    selector: { text: {} },
+  },
+  {
+    name: "default",
+    selector: { object: {} },
+  },
+  {
+    name: "selector",
+    selector: { object: {} },
```

Comment:
I understand why, but I don't like displaying a code editor here.  It makes it too complex for many users, and is unnecessary since many selectors have no options.

How about just displaying a select drop down with the different selector types?  As a first cut, could leave any non-default options only available in Yaml?

Reply:
Yeah I struggled a little bit with this too. 

I think the problem with just a dropdown is that some selectors (like `number`) have mandatory options, like min/max. So you can't really just have a "default" number selector without requiring extra input. 

It's almost like we want a `ha-selector-selector` that lets you pick a type of selector from a list, and based on what is chosen, shows an additional form of more options.

At the simplest level we could just mandate that it is a `text:` selector and not have any option at all, that might cover a lot of usecases. Don't know if that's too limited.

Reply:
Yeah, you're right.  I can try to come up with something for a selector picker as a separate PR to help.

Reply:
Yeah, we do want a `selector` selector in the future, but with all the options this can be a lot of work.

Reply:
> Yeah, we do want a `selector` selector in the future, but with all the options this can be a lot of work.

Yup... I got maybe 100 lines into it and quickly realized that. 😓

Reply:
Not sure if you were still working on this, but I did play with creating a ha-selector-selector. Styling/css is a bit rough still but it seems functional. 

I don't think we need all options of all selectors to get started with it, we can just cover some of the more common ones, and have a manual yaml option for those that fall outside the initial scope. 

Dunno if you were farther along than this, so do let me know if you want to drive it, otherwise I can continue fleshing this out. 
![selector-selector](https://github.com/home-assistant/frontend/assets/32912880/8eb9196f-8a10-4314-8d0a-b2f542e20beb)